### ISSUE TITLE
Type parameters' bounds are not preserved in generated code

### ISSUE THREAD

**Issue by gneuvill** [derive4j/hkt#29]
```java
static <A,Z> Try0<A,Z> asTry0(__<__<Try0.µ,A>,Z> hkt) {
    return (Try0<A,Z>) hkt;
  }
```
That doesn't compile. The `Z` parameter is not within its bound : it should be `Z extends Exception`.

**Comment by jbgi** [comment]
Do you want to release 0.9.3 with this fix or wait for https://github.com/derive4j/hkt/issues/14 to be resolved ? If you prefer a 0.9.3 release right away just push a `v0.9.3` tag and I will promote the resulting snapshot as release on maven central.

**Comment by gneuvill** [comment]
It can wait, as far as I am concerned ; I stumbled upon it while annotating functionaljava 's types with hkt interfaces. I intend to make a pull request (on fj) with this but not in the coming days...
